src/service/service_layer.cc

```diff
@@ -125,6 +159,50 @@ std::vector<ChirpObj> ServiceLayer::Monitor(const std::string& uname) {
   return chirps;
 }
 
+std::vector<ChirpObj> ServiceLayer::CheckTag(std::vector<ChirpObj> &chirps,const std::string& hash_tag){
+  for(std::vector<ChirpObj>::iterator it = chirps.begin(); it != chirps.end();) {
```

Comment:
I would recommend restructuring this loop.  Instead, you can write a helper that checks if a hashtag is in a single ChirpObj, and then call that on each, and add the ones that match to a return vector.